xdsl/dialects/scf.py

```diff
@@ -171,6 +191,60 @@ def get(
             regions=[body],
         )
 
+    def print(self, printer: Printer):
+        block = self.body.block
+        index, *iter_args = block.args
+        printer.print_string(" ")
+        printer.print_ssa_value(index)
+        printer.print_string(" = ")
+        printer.print_ssa_value(self.lb)
+        printer.print_string(" to ")
+        printer.print_ssa_value(self.ub)
+        printer.print_string(" step ")
+        printer.print_ssa_value(self.step)
+        printer.print_string(" ")
+        if iter_args:
+            printer.print_string("iter_args(")
+            printer.print_list(
+                zip(iter_args, self.iter_args),
+                lambda pair: print_assignment(printer, *pair),
+            )
+            printer.print_string(") -> (")
+            printer.print_list((a.type for a in iter_args), printer.print_attribute)
+            printer.print_string(") ")
+        printer.print_region(
+            self.body, print_entry_block_args=False, print_empty_block=False
+        )
+
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        index, lb = parse_assignment(parser)
+        parser.parse_characters("to")
+        ub = parser.parse_operand()
+        parser.parse_characters("step")
+        step = parser.parse_operand()
+        iter_args: list[Parser.Argument] = []
+        iter_arg_operands: list[SSAValue] = []
+        if parser.parse_optional_characters("iter_args"):
+            for iter_arg, iter_arg_operand in parser.parse_comma_separated_list(
+                Parser.Delimiter.PAREN, lambda: parse_assignment(parser)
+            ):
+                iter_args.append(iter_arg)
+                iter_arg_operands.append(iter_arg_operand)
+            parser.parse_characters("->")
+            iter_arg_types = parser.parse_comma_separated_list(
+                Parser.Delimiter.PAREN, parser.parse_attribute
+            )
+            # PR: am I doing this right? Seems like duplicated information
```

Comment:
@math-fehr @AntonLydike @PapyChacal do you know what the right move here is? Should I raise a ParseError?